rlclientlib/utility/config_utility.cc

```diff
@@ -1,138 +1,147 @@
-    std::string host;
-    std::string name;
-    std::string access_key_name;
-    std::string access_key;
-    RETURN_IF_FAIL(parse_eventhub_conn_str(conn_str, host, name, access_key_name, access_key, trace, status));
-    const auto topic = ".eventhub.";
-    cc.set(concat(cfg_root, topic, "host").c_str(),     host.c_str());
-    cc.set(concat(cfg_root, topic, "name").c_str(),     name.c_str());
-    cc.set(concat(cfg_root, topic, "keyname").c_str(), access_key_name.c_str());
-    cc.set(concat(cfg_root, topic, "key").c_str(),      access_key.c_str());
-    return error_code::success;
-  }
-
-  std::string translate(const std::map<std::string, std::string>& from_to, const std::string& from) {
-    const auto it = from_to.find(from);
-    if ( it != from_to.end() )
-      return it->second;
-    return from;
-  }
-
-  int create_from_json(const std::string& config_json, configuration& cc, i_trace* trace, api_status* status) {
-    static const std::map<std::string, std::string> legacy_translation_mapping = {
-      { "ApplicationID"             , name::APP_ID },
-      { "ModelBlobUri"              , name::MODEL_BLOB_URI },
-      { "SendHighMaterMark"         , name::INTERACTION_SEND_HIGH_WATER_MARK },
-      { "QueueMaxSize"              , name::INTERACTION_SEND_QUEUE_MAXSIZE },
-      { "SendBatchIntervalMs"       , name::INTERACTION_SEND_BATCH_INTERVAL_MS },
-      { "InitialExplorationEpsilon" , name::INITIAL_EPSILON },
-      { "ModelRefreshIntervalMs"    , name::MODEL_REFRESH_INTERVAL_MS },
-      { "QueueMode"                 , name::QUEUE_MODE } // expect either DROP or BLOCK, default is DROP
-    };
-
-    web::json::value obj;
-    try {
-      obj = json::value::parse(to_string_t(config_json));
-    }
-    catch (const web::json::json_exception& e) {
-      RETURN_ERROR_LS(trace, status, json_parse_error) << e.what();
-    }
-    auto jsonObj = obj.as_object();
-
-    for (auto const& prop_pair : jsonObj) {
-      auto prop_name = to_utf8string(prop_pair.first);
-      auto const& prop_value = prop_pair.second;
-      auto const string_value = to_utf8string(prop_value.is_string() ? prop_value.as_string() : prop_value.serialize());
-      prop_name = translate(legacy_translation_mapping, prop_name);
-
-      // Check if the current field is an EventHub connect string that needs to be parsed.
-      if (prop_name == "EventHubInteractionConnectionString") {
-        RETURN_IF_FAIL(set_eventhub_config(string_value, "interaction", cc, trace, status));
-      }
-      else if (prop_name == "EventHubObservationConnectionString") {
-        RETURN_IF_FAIL(set_eventhub_config(string_value, "observation", cc, trace, status));
-      }
-      else {
-        // Otherwise, just set the value in the config collection.
-        cc.set(prop_name.c_str(), string_value.c_str());
-      }
-    }
-
-    return error_code::success;
-  }
-
-}}}
+#include <map>
```

Comment:
Whats going on with the diff on this file?

Reply:
not sure, locally can see proper (small) diff.